src/components/Main.js

```diff
@@ -1,21 +1,43 @@
-import {useEffect, useState} from 'react'
-import {Route, Switch} from 'react-router-dom'
-import Home from '../pages/Home'
-import Run from '../pages/Run'
-import Hike from '../pages/Hike'
-import Scenic from '../pages/Scenic'
-import RunShow from '../pages/RunShow'
-import HikeShow from '../pages/HikeShow'
-import ScenicShow from '../pages/ScenicShow'
+import { useEffect, useState } from 'react';
+import { Route, Switch } from 'react-router-dom';
+import Home from '../pages/Home';
+import Run from '../pages/Run';
+import Hike from '../pages/Hike';
+import Scenic from '../pages/Scenic';
+import RunShow from '../pages/RunShow';
+import HikeShow from '../pages/HikeShow';
+import ScenicShow from '../pages/ScenicShow';
+
+const appData = {
+    runs: [
+        {
+            _id: '123',
+            location: '1600 Walnut Street, Philadelphia, PA',
+            title: 'Test Run',
+        },
+        {
+            _id: '321',
+            location: '2000 Broadway, New York, NY 10023',
+            title: 'Second Test Run',
+        },
+    ],
+};
 
 function Main(props) {
+    const [allData, setAllData] = useState({ isLoaded: false });
+    useEffect(() => {
+        setTimeout(() => {
+            setAllData({ ...appData, isLoaded: true });
+        }, 500);
+    }, [allData.isLoaded]);
+
     return (
         <main>
             <Switch>
                 <Route exact path="/">
                     <Home />
                 </Route>
-                <Route path="/run">
+                <Route exact path="/run">
```

Comment:
The show pages weren't rendering before... /run/xyz would show just /run.
Adding exact fixes that.
The same will have to be done with hike show and scenic show